src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs

```diff
@@ -82,13 +82,20 @@ private void ProcessPublishTelemetryCommand(IExecutionContext context, Dictionar
                 throw new ArgumentException(StringUtil.Loc("TelemetryCommandDataError", data, ex.Message));
             }
 
-            var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
-            var vssConnection = WorkerUtilities.GetVssConnection(context);
-            ciService.Initialize(vssConnection);
+            try
+            {
+                var ciService = HostContext.GetService<ICustomerIntelligenceServer>();
+                var vssConnection = WorkerUtilities.GetVssConnection(context);
+                ciService.Initialize(vssConnection);
 
-            var commandContext = HostContext.CreateService<IAsyncCommandContext>();
-            commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
-            commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
+                var commandContext = HostContext.CreateService<IAsyncCommandContext>();
+                commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
+                commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
```

Comment:
i don't think these 3 lines of code will throw exception, you can take them out of the try-catch
```
var commandContext = HostContext.CreateService<IAsyncCommandContext>();
commandContext.InitializeCommandContext(context, StringUtil.Loc("Telemetry"));
commandContext.Task = PublishEventsAsync(context, ciService, ciEvent);
```

Reply:
I just let it to be safe this time :)

Reply:
i would still prefer move it out. :)